### PR TITLE
LibGUI: Add highlighting to `UrlBox`

### PR DESCRIPTION
**Pull request by cammo1123**
Note: This is ported from the ladybird version #17077 
Screenshot 2023-02-09 203702

### CODE REVIEW

**Review by konradekk — APPROVED** [review]

File: Userland/Libraries/LibGUI/TextBox.cpp [hunk 1]
```diff
@@ -1,13 +1,17 @@
 /*
  * Copyright (c) 2018-2020, Andreas Kling <kling@serenityos.org>
+ * Copyright (c) 2023, Cameron Youell <cameronyouell@gmail.com>
```
Comment (AtkinsSJ) [inline Userland/Libraries/LibGUI/TextBox.cpp:3]: Nit-pick: Please put your copyright after all the existing ones.

File: Userland/Libraries/LibGUI/TextBox.cpp [hunk 2]
```diff
@@ -158,4 +172,49 @@ void UrlBox::mousedown_event(GUI::MouseEvent& event)
...
+
+    if (url.is_valid() && !is_focused()) {
+        if (url.scheme() == "http" || url.scheme() == "https" || url.scheme() == "gemini") {
+            auto host_start = url.scheme().length() + 3;
+            auto host_length = url.host().length();
+
+            // FIXME: Maybe add a generator to use https://publicsuffix.org/list/public_suffix_list.dat
+            //        for now just highlight the whole host
+
+            Gfx::TextAttributes defaultFormat;
```
Comment (AtkinsSJ) [inline Userland/Libraries/LibGUI/TextBox.cpp:188]: Use `snake_case` for variable names.

File: Userland/Libraries/LibGUI/TextBox.cpp [hunk 3]
```diff
@@ -158,4 +172,49 @@ void UrlBox::mousedown_event(GUI::MouseEvent& event)
...
+            defaultFormat.color = palette().color(Gfx::ColorRole::PlaceholderText);
+            spans.append({
+                { { 0, 0 }, { 0, host_start } },
+                defaultFormat,
+            });
+
+            Gfx::TextAttributes hostFormat;
```
Comment (AtkinsSJ) [inline Userland/Libraries/LibGUI/TextBox.cpp:195]: Same here.

File: Userland/Libraries/LibGUI/TextBox.cpp [hunk 4]
```diff
@@ -158,4 +172,49 @@ void UrlBox::mousedown_event(GUI::MouseEvent& event)
...
+                { { 0, host_start }, { 0, host_start + host_length } },
+                hostFormat,
+            });
+
+            spans.append({
+                { { 0, host_start + host_length }, { 0, text().length() } },
+                defaultFormat,
+            });
+        } else if (url.scheme() == "file") {
+            Gfx::TextAttributes schemeFormat;
```
Comment (AtkinsSJ) [inline Userland/Libraries/LibGUI/TextBox.cpp:207]: And here.

**Pull request merged**